qubekit/fragmentation/WBOFragmenter.py

```diff
@@ -0,0 +1,228 @@
+
+            # copy the fragmentation-relevant indices
+            fragment_molecule = self._create_ligand(fragment.smiles, 'fragment', bond_indices=fragment.bond_indices)
+
+            molecule.fragments.append(fragment_molecule)
+
+        return molecule
+
+    def _run(self, molecule: "Ligand", *args, **kwargs) -> "Ligand":
+        return self.run(molecule, *args, **kwargs)
+
+    def run(self, molecule: Ligand, **kwargs) -> Ligand:
+        from openff.fragmenter.depiction import depict_fragmentation_result
+        from openff.fragmenter.fragment import WBOFragmenter
+
+        # convert to an OpenFF Molecule (canonical order)
+        off_molecule: OFFMolecule = self._prepare_fragment_input(molecule)
+
+        fragmentation = WBOFragmenter().fragment(
+            off_molecule, target_bond_smarts=self.rotatable_smirks
+        )
+
+        # remove duplicates
+        fragmentation = self._deduplicate_fragments(fragmentation)
+
+        # visualise results in a file
+        depict_fragmentation_result(result=fragmentation, output_file="fragments.html")
+
+        # convert the results to QUBEKit Ligand
+        qk_molecule = self._results_to_ligand(fragmentation)
+
+        return qk_molecule
+
+    def _deduplicate_fragments(
+        self, fragmentation_result: "FragmentationResult"
+    ) -> "FragmentationResult":
+        """Remove symmetry equivalent fragments from the results."""
+        unique_fragments = []
+
+        # group by the same smiles
+        for smile, fragments in itertools.groupby(
+            fragmentation_result.fragments,
+            lambda frag: frag.molecule.to_smiles(explicit_hydrogens=False),
+        ):
+            fragments = list(fragments)
+
+            if len(fragments) == 1:
+                unique_fragments.extend(fragments)
+                continue
+
+            symmetry_groups = set()
+            for fragment in fragments:
+                bond_map = fragment.bond_indices
+                fragment_mol = fragment.molecule
+                # get the index of the atoms in the fragment
+                atom1, atom2 = self._get_atom_index(
+                    fragment_mol, bond_map[0]
+                ), self._get_atom_index(fragment_mol, bond_map[1])
+                symmetry_classes = self._rd_get_atom_symmetries(fragment_mol)
+                symmetry_group = tuple(
+                    sorted([symmetry_classes[atom1], symmetry_classes[atom2]])
+                )
+                if symmetry_group not in symmetry_groups:
+                    symmetry_groups.add(symmetry_group)
+                    unique_fragments.append(fragment)
+
+        fragmentation_result.fragments = unique_fragments
+
+        return fragmentation_result
+
+    def _rd_get_atom_symmetries(self, molecule: "Molecule") -> List[int]:
+
+        from rdkit import Chem
+
+        rd_mol = molecule.to_rdkit()
+        return list(Chem.CanonicalRankAtoms(rd_mol, breakTies=False))
```

Comment:
Is this functionally the same as `Ligand.atom_types`? if not could we make this a ligand property as it might be useful in other places something like `Ligand.atom_symmetries`?

Reply:
They are different than the attribute `atom_types`. The documentation says `If breakTies is False, this returns the symmetry class for each atom.` so an integer representing the symmetry class is assigned to each atom. I'll add the new property now.

Reply:
I confirm that the two representations are consistent. Ie atoms grouped together as X in rdkit, always belong to one group Y in qubekit.